### PR TITLE
fix: keep stdout and stderr in the error object for Docker Desktop extensions

### LINKED ISSUE #3017: microcks docker desktop extension is not working in Podman Desktop
### Bug description

https://github.com/microcks/microcks-docker-desktop-extension does not work but it's a plain UI extension so it should work

### Operating system

macOS

### Installation Method

Installer from website/GitHub releases

### Version

1.1.0

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Keep stdout ad stderr in the error object for Docker Desktop extensions
it looks like some extensions expect to find it

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop/issues/3017

### CODE REVIEW

File: packages/preload-docker-extension/src/index.ts [hunk 1]
```diff
@@ -85,10 +85,16 @@ export class DockerExtensionPreload {
       execOptions,
     );
     if (rawResult.error) {
-      throw new Error(rawResult.error);
+      const error: any = { toString: () => rawResult.stderr };
```
Comment (lstocchi) [inline packages/preload-docker-extension/src/index.ts]: Shouldn't this return the error as before? Maybe it failed before the process has been launched and stderr doesn't return anything

**Review by jeffmaury — COMMENTED** [review]
The code looks weird to me because in case of error this first method resolves an error and not an exec result and the caller will check for result.error which I don't understand where it is set

File: packages/preload-docker-extension/src/index.ts (hunk 1 above)
Comment (benoitf) [reply packages/preload-docker-extension/src/index.ts]: yes, wrong copy paste

**Comment by benoitf** [comment]
@jeffmaury should be fixed now

**Review by jeffmaury — COMMENTED** [review]
Sorry but still not getting it: if error is raised (bad command or process killed) then exit code will be 0 as error is resolved and close event won't be processed. Also I don't see when result.error is set

**Review by cdrage — APPROVED** [review]

**Comment by benoitf** [comment]
@jeffmaury looks like there was dead code. I removed the part about the .error field

File: packages/preload-docker-extension/src/index.ts (hunk 1 above)
Comment (benoitf) [reply packages/preload-docker-extension/src/index.ts]: code has been removed

**Review by lstocchi — APPROVED** [review]
LGTM

**Pull request merged**